extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthBuildTimeConfig.java

```diff
@@ -1,27 +1,108 @@
 package io.quarkus.smallrye.health.deployment;
 
 import java.util.Optional;
+import java.util.OptionalInt;
 
-import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigDocSection;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+import io.smallrye.config.WithName;
 
-@ConfigRoot(name = "smallrye-health")
-public class SmallRyeHealthBuildTimeConfig {
+@ConfigRoot
+@ConfigMapping(prefix = "quarkus.smallrye-health")
+public interface SmallRyeHealthBuildTimeConfig {
     /**
      * Activate or disable this extension. Disabling this extension means that no health related information is exposed.
      */
-    @ConfigItem(defaultValue = "true")
-    public boolean enabled;
+    @WithDefault("true")
+    boolean enabled();
 
     /**
      * Whether extensions published health check should be enabled.
      */
-    @ConfigItem(name = "extensions.enabled", defaultValueDocumentation = "true")
-    public Optional<Boolean> extensionsEnabled;
+    @WithName("extensions.enabled")
+    @WithDefault("true")
+    boolean extensionsEnabled();
 
     /**
      * Whether to include the Liveness and Readiness Health endpoints in the generated OpenAPI document
      */
-    @ConfigItem(name = "openapi.included", defaultValueDocumentation = "false")
-    public Optional<Boolean> openapiIncluded;
+    @WithName("openapi.included")
+    @WithDefault("false")
+    boolean openapiIncluded();
```

Comment:
For a weird reason, the config was split in two classes. I centralized everything in one class.